### PR TITLE
Spark: Bump to 3.4.1

### PR DESCRIPTION
**Pull request by Fokko**
Just to make sure that the CI passes

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -135,6 +135,7 @@ allprojects {
   group = "org.apache.iceberg"
   version = projectVersion
   repositories {
+    maven { url "https://repository.apache.org/content/repositories/orgapachespark-1443/" }
```
Comment (dongjoon-hyun) [inline build.gradle:138]: Thank you! Now, it's on Maven Central. :)

Comment (Fokko) [reply build.gradle]: Thanks for running the release!

**Review by singhpk234 — APPROVED** [review]
LGTM, Thanks @Fokko !

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Review by nastra — APPROVED** [review]

**Comment by nastra** [comment]
Thanks @Fokko for updating this and thanks @dongjoon-hyun and @singhpk234 for reviewing!

**Pull request merged**